### PR TITLE
Implement PostgreSQLPrivilegeHandler and add unit test

### LINKED ISSUE #10101: Implement PostgreSQLPrivilegeHandler and add unit test
Hi, community,

This issue is to `override` the `create()`, `grantAll()`,  and `diff()` of `PostregreSQLPrivilegeHandler` and also add unit tests for them in `PostgreSQLPrivilegeHandlerTest`.

### Definitions
`create()`, means creating new users (from the signature) in the database.
`grantAll()` means granting super privileges to users (from the signature) in the database.
`diff()`, means getting the user list where the users do not exist in the database.

### Reference
#10075 and #10073 are for your references.

### PR DESCRIPTION
**Pull request by totalo**
Fixes #10101 .

Changes proposed in this pull request:
- Implement PostgreSQLPrivilegeHandler and add unit test

### CODE REVIEW

File: shardingsphere-features/shardingsphere-authority/shardingsphere-authority-common/src/main/java/org/apache/shardingsphere/authority/provider/natived/builder/dialect/PostgreSQLPrivilegeHandler.java [hunk 1]
```diff
@@ -46,25 +45,56 @@
...
     public void create(final Collection<ShardingSphereUser> users, final DataSource dataSource) throws SQLException {
+        try (Connection connection = dataSource.getConnection(); Statement statement = connection.createStatement()) {
+            for (ShardingSphereUser each : users) {
+                statement.execute(getCreateUsersSQL(each));
+            }
+        }
+    }
+
+    private String getCreateUsersSQL(final ShardingSphereUser users) {
+        return String.format(CREATE_USER_SQL, users.getGrantee(), users.getPassword());
```
Comment (tristaZero) [inline shardingsphere-features/shardingsphere-authority/shardingsphere-authority-common/src/main/java/org/apache/shardingsphere/authority/provider/natived/builder/dialect/PostgreSQLPrivilegeHandler.java:82]: Hi, `user.getGrantee()` will returen `user@host` and `host` is always empty string, is that what you expect?

File: shardingsphere-features/shardingsphere-authority/shardingsphere-authority-common/src/main/java/org/apache/shardingsphere/authority/provider/natived/builder/dialect/PostgreSQLPrivilegeHandler.java [hunk 2]
```diff
@@ -130,19 +160,19 @@ private void fillRolePrivileges(final Map<ShardingSphereUser, NativePrivileges>
...
-    
+
     private void fillRolePrivileges(final Map<ShardingSphereUser, NativePrivileges> userPrivilegeMap, final ResultSet resultSet) throws SQLException {
         Optional<ShardingSphereUser> user = findShardingSphereUser(userPrivilegeMap, resultSet);
         if (user.isPresent()) {
             userPrivilegeMap.get(user.get()).getAdministrativePrivileges().getPrivileges().addAll(loadRolePrivileges(resultSet));
         }
     }
-    
+
```
Comment (tristaZero) [inline shardingsphere-features/shardingsphere-authority/shardingsphere-authority-common/src/main/java/org/apache/shardingsphere/authority/provider/natived/builder/dialect/PostgreSQLPrivilegeHandler.java:170]: Please give a look at these blank space changes.

File: shardingsphere-features/shardingsphere-authority/shardingsphere-authority-common/src/main/java/org/apache/shardingsphere/authority/provider/natived/builder/dialect/PostgreSQLPrivilegeHandler.java [hunk 3]
```diff
@@ -46,25 +45,56 @@
...
     @Override
     public Collection<ShardingSphereUser> diff(final Collection<ShardingSphereUser> users, final DataSource dataSource) throws SQLException {
-        return Collections.emptyList();
+        Collection<Grantee> grantees = new LinkedList<>();
+        try (Connection connection = dataSource.getConnection()) {
+            Statement statement = connection.createStatement();
+            try (ResultSet resultSet = statement.executeQuery(getRolePrivilegesSQL(users))) {
+                while (resultSet.next()) {
+                    grantees.add(new Grantee(resultSet.getString("rolname"), ""));
```
Comment (tristaZero) [inline shardingsphere-features/shardingsphere-authority/shardingsphere-authority-common/src/main/java/org/apache/shardingsphere/authority/provider/natived/builder/dialect/PostgreSQLPrivilegeHandler.java:65]: Does that mean `username` equals `rolename`?

**Review by tristaZero — CHANGES_REQUESTED** [review]
Hi @totalo Sorry for my coming late. Could you look at these comments?

File: shardingsphere-features/shardingsphere-authority/shardingsphere-authority-common/src/main/java/org/apache/shardingsphere/authority/provider/natived/builder/dialect/PostgreSQLPrivilegeHandler.java (hunk 1 above)
Comment (totalo) [reply shardingsphere-features/shardingsphere-authority/shardingsphere-authority-common/src/main/java/org/apache/shardingsphere/authority/provider/natived/builder/dialect/PostgreSQLPrivilegeHandler.java:82]: Yes, in postgresql, the host configuration does not seem to be tailored to the user. You can only bind the corresponding host by modifying `postgresql.conf` and `pg_hda.conf`.

File: shardingsphere-features/shardingsphere-authority/shardingsphere-authority-common/src/main/java/org/apache/shardingsphere/authority/provider/natived/builder/dialect/PostgreSQLPrivilegeHandler.java (hunk 2 above)
Comment (totalo) [reply shardingsphere-features/shardingsphere-authority/shardingsphere-authority-common/src/main/java/org/apache/shardingsphere/authority/provider/natived/builder/dialect/PostgreSQLPrivilegeHandler.java:170]: This should be a different specification. I used idea’s automatic formatting. In his specification, blank lines do not contain any characters, but in our project, blank lines seem to contain a Tab placeholder. Which one should we choose? Which way do you think is better?

File: shardingsphere-features/shardingsphere-authority/shardingsphere-authority-common/src/main/java/org/apache/shardingsphere/authority/provider/natived/builder/dialect/PostgreSQLPrivilegeHandler.java (hunk 3 above)
Comment (totalo) [reply shardingsphere-features/shardingsphere-authority/shardingsphere-authority-common/src/main/java/org/apache/shardingsphere/authority/provider/natived/builder/dialect/PostgreSQLPrivilegeHandler.java:65]: There is no concept of distinguishing users and roles in PostgreSQL, so `rolename` is equivalent to `username` here

File: shardingsphere-features/shardingsphere-authority/shardingsphere-authority-common/src/main/java/org/apache/shardingsphere/authority/provider/natived/builder/dialect/PostgreSQLPrivilegeHandler.java (hunk 2 above)
Comment (tristaZero) [reply shardingsphere-features/shardingsphere-authority/shardingsphere-authority-common/src/main/java/org/apache/shardingsphere/authority/provider/natived/builder/dialect/PostgreSQLPrivilegeHandler.java:170]: Hi, it is suggested to indent each line.
BTW, if you like, [IDEA Settings](https://shardingsphere.apache.org/community/cn/contribute/code-conduct/) is convenient to import into IDEA.

File: shardingsphere-features/shardingsphere-authority/shardingsphere-authority-common/src/main/java/org/apache/shardingsphere/authority/provider/natived/builder/dialect/PostgreSQLPrivilegeHandler.java (hunk 1 above)
Comment (tristaZero) [reply shardingsphere-features/shardingsphere-authority/shardingsphere-authority-common/src/main/java/org/apache/shardingsphere/authority/provider/natived/builder/dialect/PostgreSQLPrivilegeHandler.java:82]: Thanks for you clarification.

**Review by tristaZero — CHANGES_REQUESTED** [review]

File: shardingsphere-features/shardingsphere-authority/shardingsphere-authority-common/src/main/java/org/apache/shardingsphere/authority/provider/natived/builder/dialect/PostgreSQLPrivilegeHandler.java (hunk 2 above)
Comment (totalo) [reply shardingsphere-features/shardingsphere-authority/shardingsphere-authority-common/src/main/java/org/apache/shardingsphere/authority/provider/natived/builder/dialect/PostgreSQLPrivilegeHandler.java:170]: Okay, thank you, I'm happy to do this.

**Review by tristaZero — APPROVED** [review]
@totalo I can see your being-careful from this PR, great.

**Comment by totalo** [comment]
> @totalo I can see your being-careful from this PR, great.

hhhh , I am happy to do these things. BTW, I want to participate in contributing more things. Can you give me some more directions?

**Comment by tristaZero** [comment]
Hey, Surely. 
I will update here once some issues are ready for you.

**Comment by totalo** [comment]
> Hey, Surely.
> I will update here once some issues are ready for you.

Thank you very much.

**Pull request merged**